src/components/state-lookup.ts

```diff
@@ -191,9 +187,22 @@ export class StateLookup {
         }
 
         // blunt update
-        if (!r.events[event.type]) {
-            r.events[event.type] = {};
+        this.insertEvent(r, event);
+    }
+
+    private insertEvent(roomSet: StateLookupRoom, event: StateLookupEvent) {
+        if (typeof event.content !== "object") {
+            // Reject - unexpected content type
+            return;
+        }
+        if (!event.type || !event.state_key) {
```

Comment:
Just for learning something: This broke the tests. It's not testing if these keys are missing (as claimed by the line below), but if they are truethy. The tests use empty strings, which aren't truethy.

Reply:
Fixed in https://github.com/matrix-org/matrix-appservice-bridge/pull/185/commits/deb2f38ab33b74567bba6be879440da1d26f5a4b